docs/en/pact-reference.md

```diff
@@ -243,6 +243,30 @@ signers:
 type: exec
 ```
 
+### WebAuthn Signature Format
+
+The Pact REST API can accept transactions signed by the WebAuthn protocol.
+
+A web client can make use of this scheme to allow users to authenticate and
+sign pact transactions without manually managing private keys. The client must:
+
+ - Modify the `Signer`, it must have `scheme` set to `"WebAuthn"`.
+ - Construct the challenge as: `base64( blake2b ( CommandPayload ))`
+ - Issue a JavaScript `navigator.credentials.get` with this challenge.
+ - Construct a `Signature` from the credentials response.
+
+The "signature" to pass to Pact (in the `sigs` field) is a sringified JSON
+object composed of pieces taken from the browser's credential response. Some of
+the response fields must be converted from the Base64URL alphabet to Base64:
+
+```
+{
```

Comment:
Is it possible to have this interface changed from:
```
"sigs": [{
  "sig": JSON.stringify({
      "signature": toBase64(response.signature),
      "authenticatorData": toBase64(response.authenticatorData),
      "clientDataJSON": toBase64URL(response.clientDataJSON)
    })
 }]
 ```
to:
```
"sigs": [{
  "sig": toBase64(response.signature),
  "authenticatorData": toBase64(response.authenticatorData),
  "clientDataJSON": toBase64URL(response.clientDataJSON)
}]
```

Reply:
That would be a forking change. I think we could do it if we wanted, but we would want to batch related tweaks to the format together and push them out all at once if possible, to avoid introducing too many forks.

I believe @emilypi and @edmundnoble also had ideas about moving the non-sig data out of the signature and into some other field of the request? If this is the case, our webauthn sig would be reduced to just the browser's "signature".

Reply:
@imalsogreg maybe by using a small lib we could simplify the docs for the JS usage:

```ts
import { startAuthentication, base64URLStringToBuffer } from "@simplewebauthn/browser";

const res = await startAuthentication({
  challenge: tx.hash,
  rpId: window.location.hostname,
  allowCredentials: [{ id: cid, type: "public-key" }], // optional
});

const signature = Buffer.from(
  base64URLStringToBuffer(response.signature)
).toString("base64");
const authenticatorData = Buffer.from(
  base64URLStringToBuffer(response.authenticatorData)
).toString("base64");
const clientDataJSON = Buffer.from(
  base64URLStringToBuffer(response.clientDataJSON)
).toString("base64");

const sig = {
  sig: JSON.stringify({
    signature,
    authenticatorData,
    clientDataJSON,
  })
}
```

Reply:
Moved to a ticket.